packages/appium/lib/appium.js

```diff
@@ -535,6 +887,30 @@ class AppiumDriver extends DriverCore {
     }
   }
 
+  /**
+   * @param {string} sessionId
+   */
+  cleanupBidiSockets(sessionId) {
+    // clean up any bidi sockets associated with session
+    if (this.bidiSockets[sessionId]) {
+      try {
+        this.log.debug(`Closing bidi socket(s) associated with session ${sessionId}`);
+        for (const ws of this.bidiSockets[sessionId]) {
+          ws.close();
```

Comment:
not sure if we want to use error code and reason to inform the client about the close reason